include/arch/arm/aarch32/cortex_m/scripts/linker.ld

```diff
@@ -446,6 +449,35 @@ GROUP_START(DTCM)
 GROUP_END(DTCM)
 #endif
 
+#ifdef CONFIG_ETH_STM32_HAL_USE_SRAM3
```

Comment:
This could also be an #ifdef directive based on DT.

Reply:
Done.